New package: ROS2 v0.1.5

JuliaRegistrator opened the pull request:
- Registering package: ROS2
- Repository: https://github.com/Michi-Tsubaki/ROS2.jl
- Created by: @Michi-Tsubaki
- Version: v0.1.5
- Commit: bb362b0d266519215a2ad396d0bb5876d1bdf0c4
- Reviewed by: @Michi-Tsubaki
- Reference: https://github.com/Michi-Tsubaki/ROS2.jl/issues/6#issuecomment-2634591295
- Description: Robot Operating tool using Julia
<!-- bf0c69308befbd3ccf2cc956ac8a4671113b158fd87a6d4e741b0173dab051a48f7dc43e593ffff4357bc647a3a83598ef5d13a2899ca805b9ec225fb5aa7d2b675ec081d472063577d244471d3e612e8ae65f08b7e684eb99799b6e85f6f1e7148533c56cdcb092dc3632f3f81a2259673fc8ef0a242e5179897b47904084842fa4351eb532d8a0244befda77ddee536f097f7a056e47065676e43d6d8809cc4e060143773bb0ea7a1a8ec902099368 -->

Michi-Tsubaki commented:
Same Problem happens https://github.com/JuliaRegistries/General/pull/124282 . Please dismiss this PR. Very sorry.

Michi-Tsubaki commented:
@goerz
Sorry for making several PRs (In https://github.com/Michi-Tsubaki/ROS2.jl/issues/6, it cannot triggerd in same version tag) 
Could you please close PR for v0.1.4(https://github.com/JuliaRegistries/General/pull/124282) .

As shown below ↓

> . [AutoMerge Guidelines](https://juliaregistries.github.io/RegistryCI.jl/stable/guidelines/) which are not met ❌
Name does not meet all of the following: starts with an upper-case letter, ASCII alphanumerics only, not all letters are upper-case.
Name is not at least 5 characters long
Package name similar to 3 existing packages.
Similar to ROS. Damerau-Levenshtein distance 1 is at or below cutoff of 2. Damerau-Levenshtein distance 1 between lowercased names is at or below cutoff of 1.
Similar to BOSS. Damerau-Levenshtein distance 2 is at or below cutoff of 2. Normalized visual distance 1.97 is at or below cutoff of 2.50.
Similar to ROOT. Damerau-Levenshtein distance 2 is at or below cutoff of 2.

According to CI log
>  Warning: ROS2 initialization deferred: ProcessFailedException(Base.Process[Process(setenv(`/tmp/jl_hH9yG4/conda/3/x86_64/bin/conda install -y rclpy`,["PATH=/opt/hostedtoolcache/julia/1.11.3/x64/bin:/snap/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin", "CONDA_PREFIX=/tmp/jl_hH9yG4/conda/3/x86_64", "***", "DISPLAY=:99", "OPENBLAS_DEFAULT_NUM_THREADS=1", "JULIA_REGISTRYCI_AUTOMERGE=true", "XAUTHORITY=/tmp/xvfb-run.qnHAeW/Xauthority", "HOME=/home/runner", "JULIA_PKG_PRECOMPILE_AUTO=0", "CONDARC=/tmp/jl_hH9yG4/conda/3/x86_64/condarc-julia.yml", "JULIA_DEPOT_PATH=/tmp/jl_hH9yG4", "OPENBLAS_MAIN_FREE=1", "R_HOME=*", "PYTHONIOENCODING=UTF-8"]), ProcessExited(1))])
└ @ ROS2 /tmp/jl_hH9yG4/packages/ROS2/SK1VV/src/ROS2.jl:32
[ Info: Successfully `import`ed package
[ Info: Suc

ROS2 initialize problem = import problem has pass CI, by making  ERROR --> Warning

Michi-Tsubaki commented:
Sorry, last change does not work... around ROS2 initialization, please wait merge...

JuliaRegistrator committed:
New package: ROS2 v0.1.5

UUID: 493365b4-7e18-460b-8b90-ec3e7f80d3eb
Repo: https://github.com/Michi-Tsubaki/ROS2.jl.git
Tree: 52d397800cbda73dc27bfdb4ff628654e405ba65

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

Michi-Tsubaki commented:
[noblock]

goerz commented:
> Sorry for making several PRs (In https://github.com/Michi-Tsubaki/ROS2.jl/issues/6, it cannot triggerd in same version tag)

This is why you should not create version tags manually. The normal workflow in Julia (which is very unusual compared to how most other systems handle it) is that you submit a particular commit for registration. Once that is merged, the [TagBot](https://github.com/JuliaRegistries/TagBot) in your repo is triggered, and that is what creates the tag for the release. This is important, because you might run into problems during the registration (such as in https://github.com/JuliaRegistries/General/pull/124282, but potentially also for releases after the initial registration), and you can _update_ a registration to fix these problems while keeping the version number the same. Creating the tag _after_ everything is fixed ensures that the tag is pointing to the correct commit that was actually registered.

BTW, I never really understood this "registration via comment on an issue". I know there's an older tutorial from Chris floating around that recommends this, but personally, it makes much more sense to me to just navigate to the commit you want to submit for registration, and comment `@JuliaRegistrator register` on that commit. If something needs to be fixed, you just make another commit, or even rewrite/force-push, and just comment again on the new commit to update the existing registration PR.

Michi-Tsubaki commented:
@goerz Can you put [noblock] in the comment above? I feel like it is triggering the lock, maybe.

Michi-Tsubaki commented:
OK, I see. Thank you.
